src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,76 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo | null;
+  setSelectedTodo: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  const onCloseHandler = (currentTodo: Todo | null) => {
+    setSelectedTodo(currentTodo);
+    setUser(null);
+  };
+
+  useEffect(() => {
+    if (selectedTodo) {
+      getUser(selectedTodo.userId).then(setUser);
```

Comment:
It's good that you're using promises to fetch data from the API, but it would be better if you handled possible errors. If the promise rejects, the error will be unhandled and it may break your application. You can add a `.catch` block to handle possible errors.